tests/keras/layers/test_convolutional.py

```diff
@@ -61,27 +61,32 @@ def test_convolution_2d():
 
     for border_mode in ['valid', 'same']:
         for subsample in [(1, 1), (2, 2)]:
-            if border_mode == 'same' and subsample != (1, 1):
-                continue
-
-            layer_test(convolutional.Convolution2D,
-                       kwargs={'nb_filter': nb_filter,
-                               'nb_row': 3,
-                               'nb_col': 3,
-                               'border_mode': border_mode,
-                               'subsample': subsample},
-                       input_shape=(nb_samples, stack_size, nb_row, nb_col))
-
-            layer_test(convolutional.Convolution2D,
-                       kwargs={'nb_filter': nb_filter,
-                               'nb_row': 3,
-                               'nb_col': 3,
-                               'border_mode': border_mode,
-                               'W_regularizer': 'l2',
-                               'b_regularizer': 'l2',
-                               'activity_regularizer': 'activity_l2',
-                               'subsample': subsample},
-                       input_shape=(nb_samples, stack_size, nb_row, nb_col))
+            for filter_dilation in [(1, 1), (2, 2)]:
+                if border_mode == 'same' and subsample != (1, 1):
+                    continue
+                if subsample != (1, 1) and filter_dilation != (1, 1):
```

Comment:
Also thanks for your review.
I think it's good to have a better coverage, but I am not sure if it's appropriate to make different test code for different backend, there is no example in the current test code which have different test for different backend (there is one, but it skip the entire test function for tensorflow). I would like to hear other's opinion on this before doing that.